docs/elasticsearch-spec.asciidoc

```diff
@@ -359,64 +359,51 @@ The init container inherits the image of the main container image if one is not
-3. Once data is migrated, remove the first 16GB node
-4. Follow the same steps for the 2 other 16GB nodes
+==== Specifying changeBudget
+For both `maxSurge` and `maxUnavailable` you can specify the following values:
 
-The cluster health stays green during the entire process.
-By default, only one extra node can be added on top of the expected ones. In the example above, a 3-nodes cluster may temporarily be composed of 4 nodes while data migration is in progress.
+* `null` - the default value will be used,
+* non-negative - the value will be used as is,
+* negative - the value will be unbounded.
 
-This behaviour can be controlled through the `changeBudget` section of the cluster specification `updateStrategy`. If not specified, it defaults to the following:
+==== Default behavior
+When `updateStrategy` is not present in the spec, it defaults to the following:
 
 [source,yaml]
 ----
 spec:
   updateStrategy:
     changeBudget:
-      maxSurge: 1
-      maxUnavailable: 0
+      maxSurge: -1
+      maxUnavailable: 1
 ----
+`maxSurge` is unbounded. This results in immediate creation of all missing Pods that best practices allow for at the given time.
+`maxUnavailable` defaults to `1`. This assures that cluster will have at most one unavailable Pod at any given time.
 
-* `maxSurge` specifies the number of Pods that can be added to the cluster, on top of the desired number of nodes in the specification during cluster updates
-* `maxUnavailable` specifies the number of Pods that can be made unavailable during cluster updates
-
-The default of `maxSurge: 1; maxUnavailable: 0` spins up an additional Elasticsearch node during cluster updates.
-It is possible to speed up cluster topology changes by increasing `maxSurge`. For example, setting `maxSurge: 3` would allow 3 new nodes to be created while the original 3 migrate data in parallel.
-The cluster would then temporarily have 6 nodes.
-
-Setting `maxSurge` to 0 and `maxUnavailable` to a positive value only allows a maximum number of Pods to exist on the Kubernetes cluster.
-For example, `maxSurge: 0; maxUnavailable: 1` would perform the 3 nodes upgrade this way:
-
-1. Migrate data away from the first 16GB node
-2. Once data is migrated, remove the 16GB node: the cluster temporarily has 2 nodes
-3. Add a new 32GB node: the cluster grows to 3 nodes
-4. Follow the same steps for the 2 other 16GB nodes
-
-Even if a `changeBudget` is specified, ECK makes sure that some invariants are maintained while a mutation is in progress. In the cluster, there must be at least:
-
-* One master node alive
-* One data node alive
 
-So under certain circumstances ECK ignores the change budget. For example, a safe migration from a 1-node cluster to another 1-node cluster can only be done by temporarily setting up a 2-nodes cluster.
+==== Caveats
+* Setting both `maxSurge` and `maxUnavailable` to `0` will prevent the operator from performing any update that needs to bring down or create a new Pod.
```

Comment:
With both `maxSurge` and `maxUnavailable` set to `0`, the operator cannot bring down an existing Pod nor create a new Pod.